### ISSUE TITLE
Database Schema

### ISSUE THREAD

**Issue by ylt** [WritheM/bonsai#25]
Not finalised, just a quick summary of the planned database schema:

User:
- id
- username
- slug
- language
- email
- gRole

Room:
- id
- owner (FK: user)
- name
- summary
- description

RoomRolePrivs:
- role_id (FK: RoomRole)
- privilege

RoomRole:
- id
- room_id (FK: room)
- priority (higher values are sorted first)
- name
- description
- icon

RoomStaff: (PK: room_id, user_id)
- room_id (FK: room)
- user_id (FK: user)
- role_id (FK: RoomRole)

Media:
- id
- user_id (FK: user)
- duration
- mediatype (1: yt, 2: sc, 3: bandcamp, 4: special)

Play:
- id
- room_id (FK: room)
- media_id (FK: media)
- added_by (FK: user)
- start_time

Queue:
- id
- room_id
- owner_id (FK: user)
- room_id

Chat:
- id
- room_id (FK: room)
- user_id (FK: user)
- message
- style (0: normal, 1: emote, maybe other styles?)

**Comment by gkampjes** [comment]
Not sure we need style in the chat table. Wouldn't emotes/emojis be done using markup/down?

Upon saying that, you could use style for admin messages etc.

Also, won't media need a url?

**Comment by ylt** [comment]
Yeah, style will be used for admin messages, etc.. Maybe will use it for other stuff but undecided about that...

And yep, media needs a URL.. This was just a rushed schema so that we have a rough idea of what we're aiming for, wasn't meant to be perfect ;)
